src/config.rs

```diff
@@ -0,0 +1,1235 @@
+                max_completed_connections: 2048,
+                max_incomplete_connections: 64,
+                max_connections_per_user: 256,
+                max_pending_service_starts: 512,
+                max_names_per_connection: 512,
+                max_match_rules_per_connection: 512,
+                max_replies_per_connection: 128,
+                ..Default::default()
+            },
+            None => {
+                // TODO: consider panicking or returning an Error or something
+                Limits::default()
+            }
+        }
+    }
+
+    pub fn parse(s: &str) -> Result<Self> {
+        // validate that our DOCTYPE and root element are correct
+        let mut reader = Reader::from_reader(s.as_bytes());
+        let mut has_dtd = false;
+        loop {
+            match reader.read_event()? {
+                Event::DocType(bytes_text) => {
+                    has_dtd = true;
+                    let content = bytes_text.unescape()?;
+                    // this approach does throw away extra whitespace within quoted strings,
+                    // which means we do allow through some unexpected DOCTYPEs
+                    // TODO: consider whether added complexity is worth 100% strictness for DOCTYPEs
+                    let dtd_parts: Vec<&str> = content.as_ref().split_whitespace().collect();
+                    if dtd_parts != EXPECTED_DOCTYPE_PARTS {
+                        assert_eq!(dtd_parts, EXPECTED_DOCTYPE_PARTS);
+                        return Err(Error::msg(
+                            "incorrect/incomplete `<!DOCTYPE ...> declaration`",
+                        ));
+                    }
+                    // we currently don't bother with the case of multiple DTDs
+                }
+                Event::Start(bytes_start) => {
+                    if !has_dtd {
+                        return Err(Error::msg(
+                            "must specify `<!DOCTYPE ...>` before root element",
+                        ));
+                    }
+                    if bytes_start.name().as_ref() != b"busconfig" {
+                        return Err(Error::msg("root element must be `<busconfig>`"));
+                    }
+                    break;
+                }
+                Event::Eof => break,
+                _ => {}
+            }
+        }
+
+        quick_xml::de::from_str(s).map_err(Error::from)
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum BusType {
+    /// The DBUS_STARTER_BUS_TYPE environment variable will be set to "session",
+    /// and the DBUS_SESSION_BUS_ADDRESS environment variable will be set to the address of the
+    /// session bus.
+    Session,
+    /// The DBUS_STARTER_BUS_TYPE environment variable will be set to "system",
+    /// and the DBUS_SYSTEM_BUS_ADDRESS environment variable will be set to the address of the
+    /// system bus.
```

Comment:
I know you got these docs from the spec but I'm not sure what it really means. None of the existing brokers do this (I don't even remember seeing `DBUS_STARTER_BUS_TYPE` defined in any env (even the `DBUS_SYSTEM_BUS_ADDRESS` is never defined in a session). 

Looking into the dbus-daemon code, it seems to be setting and then using this env but it all looks very internal. I don't think we need to do any of that. I'd just remove the doc comments here.